KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,348 +1,656 @@
-                        ref14III.append(index)
-        if("\\subsection*{References}" in word) and (canAdd):
+                if (processed_word in subunit or specially_formatted_word in subunit) and (chaptercheck in unit) and len(processed_word) == len(subunit) or\
+                ("Pseudo Jacobi" in processed_word and "Pseudo Jacobi (or Routh-Romanovski)" in subunit):
+                    add_next_section = True
+                    if chapticker == 9:
+                        ref9_3.append(index)
+                    elif chapticker == 14:
+                        ref14_3.append(index)
+                    special_detector = 0
+        if "\\subsection*{References}" in word and add_next_section:
             # Appends valid locations
             references.append(index)
-            if chapticker == 0:
-                ref9II.append(index)
-                ref9III.append(index)
-            elif chapticker == 1:
-                ref14II.append(index)
-                ref14III.append(index)
-            canAdd = False
-        if ("subsection*{" in word and "References" not in word):
-            if chapticker == 0:
-                ref9III.append(index)
-            elif chapticker == 1:
-                ref14III.append(index)
-    print(ref9II)
-    print (ref14II)
-    print(ref9III)
-    print(ref14III)
-    return references
-
-
-def referencePlacer(chap, references, p, kls, refII, refIII, specref):
+            if chapticker == 9:
+                ref9_3.append(index)
+            elif chapticker == 14:
+                ref14_3.append(index)
+            add_next_section = False
+        if "subsection*{" in word and "References" not in word:
+            if chapticker == 9:
+                ref9_3.append(index)
+            elif chapticker == 14:
+                ref14_3.append(index)
+        if "\\section{" in word and special_detector == 1 and chaptercheck == "14":
+            w2 = word[word.find("{") + 1: word.find("}")]
+            if "Bessel" not in w2:
+                ref14_3.append(index)
+
+    return references, ref9_3, ref14_3
+
+
+def reference_placer(chap, references, p, chapticker2):
+    """
+    Places identified additions from the addendum list into the chapter list. The list returned will be turned into a
+    string and written into the new text document
+
+    :param chap: LaTeX Chapter receiving the additions, with each line an element of a list.
+    :param references: List containing references to where subsections begin/end
+    :param p: List containing additions to be added.
+    :param chapticker2: Which chapter is being searched (9 or 14).
+    :return: Returns the chapter, complete with additions
+    """
     # count is used to represent the values in count
     count = 0
     # Tells which chapter it's on
     designator = 0
-    if chapticker2 == 0:
+    if chapticker2 == 9:
         designator = "9."
-    elif chapticker2 == 1:
+    elif chapticker2 == 14:
         designator = "14."
```

Comment:
why not instead of doing the comparison, do `designator = str(chapticker2) + "."`

Also why is it called chapticker2 instead of just chapticker